src/macaron/slsa_analyzer/package_registry/maven_central_registry.py

```diff
@@ -236,3 +239,84 @@ def find_publish_timestamp(self, purl: str) -> datetime:
                 raise InvalidHTTPResponseError(f"The timestamp returned by {url} is invalid") from error
 
         raise InvalidHTTPResponseError(f"Invalid response from Maven central for {url}.")
+
+    @staticmethod
+    def get_artifact_file_name(purl: PackageURL) -> str | None:
+        """Return the artifact file name of the passed PURL based on the Maven registry standard.
+
+        Parameters
+        ----------
+        purl: PackageURL
+            The PURL of the artifact.
+
+        Returns
+        -------
+        str | None
+            The artifact file name, or None if invalid.
+        """
+        if not purl.version:
+            return None
+
+        return purl.name + "-" + purl.version + ".jar"
+
+    def get_artifact_hash(self, purl: PackageURL, hash_algorithm: Any) -> str | None:
+        """Return the hash of the artifact found by the passed purl relevant to the registry's URL.
+
+        Parameters
+        ----------
+        purl: PackageURL
+            The purl of the artifact.
+        hash_algorithm: Any
+            The hash algorithm to use.
+
+        Returns
+        -------
+        str | None
+            The hash of the artifact, or None if not found.
+        """
+        if not purl.namespace:
+            return None
+
+        file_name = MavenCentralRegistry.get_artifact_file_name(purl)
```

Comment:
Since we just want to get the "main" artifact name from a Maven PURL, this method should not be specific to Maven Central. We also have a [`maven.py` ](https://github.com/oracle/macaron/blob/915-discover-github-attestations/src/macaron/artifact/maven.py)module that might be more suitable for this method.